src/i18n/arabic.rs

```diff
@@ -93,26 +93,34 @@ fn get_contextual_form_of_char(prev: char, next: char, c: char) -> Option<u32> {
     // The reference index now takes into account the special Lam-Alif combination.
     let ref_index = if is_la { next_cp } else { cp } - AR_START;
 
-    let target = ((((is_lapl || is_arabic_letter(next_cp)) && is_linking_type(cp)) as u32) << 1)
-        | is_linking_type(prev_cp) as u32;
+    let target = if is_lapl {
+        if is_linking_type(prev_cp) {
+            5 // Use final form
+        } else {
+            4 // Use isolated form
+        }
+    } else {
+        ((((is_arabic_letter(next_cp)) && is_linking_type(cp)) as u32) << 1)
+            | is_linking_type(prev_cp) as u32
+    };
```

Comment:
**Bug**:  
Incorrect fixed index used for `la` in `is_la` condition, not considering the absence of a previous linking character.

**Fix**:  
Updated logic to dynamically select the target index for `la`:
- Uses the final form (index 5) if the previous character is of a linking type.
- Uses the isolated form (index 4) otherwise, since `la` does not differentiate between initial and isolated forms.

**Request**:  
Please let me know if there is a better way for calculating the target index.